### ISSUE TITLE
YOLOv8n validation on DOTA v2.0 dataset

### ISSUE THREAD

**Issue by HornGate** [ultralytics/ultralytics#19697]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello! When I use the YOLOv8n model to save TXT results (predictions_merged_txt) on the DOTA v2.0 validation and test sets, I frequently encounter out-of-memory (OOM) errors. How can I resolve this issue?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @HornGate, thank you for reaching out and for your interest in Ultralytics 🚀! 

We recommend checking out our [Docs](https://docs.ultralytics.com/) for guidance on using YOLO models, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. Many common questions and troubleshooting steps are covered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that demonstrates the issue. This will help us investigate and address the problem more effectively.

If your question relates to custom training ❓, please share details such as dataset examples, training logs, and the steps you've followed. Additionally, refer to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure optimal performance.

### Upgrade

Please ensure you are using the latest version of the `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). You can upgrade with:

```bash
pip install -U ultralytics
```

### Environments

YOLO models can be run in various verified environments with all dependencies preinstalled:

- **Notebooks** with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Community Support

For real-time discussions, join our [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth conversations, explore [Discourse](https://community.ultralytics.com/). And don't forget to share insights or ask questions on our [Subreddit](https://reddit.com/r/Ultralytics).

### Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing, ensuring the stability of YOLO across macOS, Windows, and Ubuntu.

---

This is an automated response to guide you towards helpful resources and solutions. An Ultralytics engineer will review your issue and assist you further soon. Thank you for your patience and for being a part of the Ultralytics community! 😊

**Comment by glenn-jocher** [comment]
@HornGate for OOM issues with DOTA v2.0 validation using YOLOv8n-OBB, we recommend:

1. **Reduce batch size** with `batch=...` during validation
2. **Decrease image size** using `imgsz=...` (test with 1024 or lower)
3. **Adjust NMS IoU threshold** in the merging step (default 0.3) via code modification in the `eval_json` method shown in the Oriented Detection Validation documentation

Ensure you're using the latest PyTorch and Ultralytics versions for memory optimizations. For hardware-specific guidance, please share your device specs.
